tapir/coop/services/ResignMemberService.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+
+from tapir.accounts.models import TapirUser
+from tapir.coop.models import ResignedMembership, ShareOwnership
+from tapir.coop.forms import MembershipCancelForm
+from tapir.shifts.models import (
+    ShiftAttendanceTemplate,
+    ShiftAttendance,
+    ShiftAccountEntry,
+)
+from tapir.shifts.services.frozen_status_service import FrozenStatusService
+from tapir.shifts.models import ShiftUserData, ShiftAttendanceMode
+
+from tapir.utils.shortcuts import get_timezone_aware_datetime
+
+
+class ResignMemberService:
+    @staticmethod
+    def update_shifts_and_shares(
+        form: MembershipCancelForm, member: ResignedMembership
+    ):
+        tapir_user: TapirUser = getattr(member.share_owner, "user", None)
+        if not tapir_user:
+            print("Couldn't find an existing Tapir user.")
+            return
+        shares = ShareOwnership.objects.filter(share_owner=member.share_owner)
+        if (
+            member.willing_to_gift_shares_to_coop
+            or member.transfering_shares_to is not None
+        ):
+            for share in shares:
+                if share.is_fully_paid:
```

Comment:
`share.is_fully_paid` is deprecated: we don't use `ShareOwnership.amount_paid` anymore. 
Instead you can use `ShareOwner.get_currently_paid_amount` and `get_total_expected_payment`

Reply:
`ShareOwnership.amount_paid` is not filled anymore, since we introduced payment tracking with `IncomingPayment` all shares have `amount_paid == 0` even if they are paid. So we can't use that field at all.

That means we don't have the information about which specific share is paid, only if everything has been paid or not.
I think we can block the entire resignation if not everything is paid?

Reply:
Done